storage/oasis/nodeapi/api.go

```diff
@@ -79,36 +80,41 @@ type TxError consensusResults.Error
 // the indexer actively inspects (as opposed to just logging them) to perform
 // dead reckoning or similar state updates.
 type Event struct {
-	Height int64
-	TxHash hash.Hash
-
-	// The fine-grained Event struct directly from oasis-core; corresponds to
-	// at most one of the fields below (StakingTransfer, StakingBurn, etc.).
-	Body interface{}
+	Height int64     `json:"height"`
+	TxHash hash.Hash `json:"tx_hash"`
+
+	// The body of the Event struct as it was received from oasis-core. For most
+	// event types, a summary of the event (containing only indexer-relevant fields)
+	// will be present in one of the fields below (StakingTransfer, StakingBurn, etc.).
+	// For event types that the indexer doesn't process beyond logging, only this
+	// field will be populated.
+	// We convert to JSON and effectively erase the type here in order to decouple
+	// oasis-core types (which vary between versions) from the indexer.
+	RawBodyJSON json.RawMessage `json:"raw_body"`
 
 	// Called "Kind" in oasis-core but "Type" in indexer APIs and DBs.
-	Type apiTypes.ConsensusEventType
-
-	StakingTransfer             *TransferEvent
-	StakingBurn                 *BurnEvent
-	StakingAddEscrow            *AddEscrowEvent
-	StakingTakeEscrow           *TakeEscrowEvent
-	StakingEscrowDebondingStart *DebondingStartEscrowEvent
-	StakingReclaimEscrow        *ReclaimEscrowEvent
-	StakingDebondingStart       *DebondingStartEscrowEvent // Available starting in Damask.
-	StakingAllowanceChange      *AllowanceChangeEvent
-
-	RegistryRuntimeRegistered *RuntimeRegisteredEvent
-	RegistryEntity            *EntityEvent
-	RegistryNode              *NodeEvent
-	RegistryNodeUnfrozen      *NodeUnfrozenEvent
-
-	RoothashExecutorCommitted *ExecutorCommittedEvent
-
-	GovernanceProposalSubmitted *ProposalSubmittedEvent
-	GovernanceProposalExecuted  *ProposalExecutedEvent
-	GovernanceProposalFinalized *ProposalFinalizedEvent
-	GovernanceVote              *VoteEvent
+	Type apiTypes.ConsensusEventType `json:"type"`
```

Comment:
If we no longer need the json annotations, I'let's remove them to keep things simple.
(I think they were not strictly needed even when we were serializing into JSON.)